### PR TITLE
Add support for rerun tag in TKG

### PR DESCRIPTION
**Pull request by renfeiw**
- update the parsing logic
- update the util gen logic to generate IGNORE_ON_RERUN
- add corresponding test case

### CODE REVIEW

**Comment by renfeiw** [comment]
run test-tkg

**Comment by renfeiw** [comment]
Grinder_TKG/1894 - Grinder_TKG/1896

**Comment by renfeiw** [comment]
run test-tkg

**Comment by renfeiw** [comment]
run tkg-test

File: examples/rerun/playlist.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+<playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../TKG/playlist.xsd">
+	<test>
+		<testCaseName>testRerun</testCaseName>
+		<command>echo "test rerun"; \
+	$(TEST_STATUS)</command>
```
Comment (karianna) [inline examples/rerun/playlist.xml:19]: whitespace?  Same below

File: scripts/testTKG/test_rerun.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import subprocess
+from utils import *
+
+def run_test():
```
Comment (karianna) [inline scripts/testTKG/test_rerun.py:18]: Worth a comment on this and the other function on 'what they do'

**Review by karianna — CHANGES_REQUESTED** [review]
Can the design of this be changed to provide a rerun list as opposed to a 'notRerun' list - the logic seems backwards/more complex than it needs to be.

**Comment by llxia** [comment]
> Can the design of this be changed to provide a rerun list as opposed to a 'notRerun' list - the logic seems backwards/more complex than it needs to be.

By default, all failed tests will be rerun if the auto-rerun feature is enabled. This PR is to handle the special case, where some tests have known failures and do not need to be rerun automatically. See https://github.com/adoptium/aqa-tests/issues/4379

**Comment by smlambert** [comment]
For clarification, the 'notRerun' would be more aptly named 'ignoreOnRerun' or 'notToBeRerun'.

File: examples/rerun/playlist.xml (hunk 1 above)
Comment (renfeiw) [reply examples/rerun/playlist.xml:19]: The whitespace is expected format for command in playlist.xml file.

File: scripts/testTKG/test_rerun.py (hunk 2 above)
Comment (renfeiw) [reply scripts/testTKG/test_rerun.py:19]: Updated, thank you!

**Comment by renfeiw** [comment]
> For clarification, the 'notRerun' would be more aptly named 'ignoreOnRerun' or 'notToBeRerun'.

I have changed NOT_RERUN_LIST to IGNORE_ON_RERUN.

**Comment by renfeiw** [comment]
run tkg-test

**Review by llxia — APPROVED** [review]
LGTM

File: src/org/testKitGen/TestInfoParser.java [hunk 3]
```diff
@@ -112,6 +112,19 @@ public TestInfo parse() {
...
 		}
 
+		String rerun = getImmediateChildContent(testEle, "rerun");
+		if (rerun != null) {
+			if (rerun.equals("true")) {
+				ti.setRerun(true);
+			} else if (rerun.equals("false")) {
+				ti.setRerun(false);
+			} else {
+				System.err.println("Error: Please provide a valid rerun parameter in test " + ti.getTestCaseName() + ". The valid string is [true|false].");
```
Comment (karianna) [inline src/org/testKitGen/TestInfoParser.java:122]: Do we not have as logger for this?

**Review by karianna — COMMENTED** [review]
Can we change the notRerun wording to ignoreOnRerun wording?

File: src/org/testKitGen/TestInfoParser.java (hunk 3 above)
Comment (renfeiw) [reply src/org/testKitGen/TestInfoParser.java:122]: No, we don't have logger for now. It can be an improvement to add something like log4j in the future.

**Comment by renfeiw** [comment]
run tkg-test

**Comment by renfeiw** [comment]
run tkg-test

**Comment by renfeiw** [comment]
> Can we change the notRerun wording to ignoreOnRerun wording?

I have changed all notRerun to ignoreOnRerun.

File: src/org/testKitGen/TestInfoParser.java (hunk 3 above)
Comment (smlambert) [reply src/org/testKitGen/TestInfoParser.java:123]: For the record, loggers are extra dependencies that we have purposefully tried to keep to the very minimum for AQAvit work, as there is overhead with downloading/maintaining/updating each new dependency added.

Comment (karianna) [reply src/org/testKitGen/TestInfoParser.java:123]: At some point dumping to console only is going to restrict debugging, but that's a problem for another day :-)

**Review by karianna — APPROVED** [review]

**Pull request merged**